src/App.tsx

```diff
@@ -21,75 +24,130 @@ enum SortType {
   LENGTH,
 }
 
-type ReorderOptions = {
-  sortType: SortType,
+type State = {
   isReversed: boolean,
+  sortType: SortType,
 };
 
-// Use this function in the render to prepare goods
-export function getReorderedGoods(
+export function implementButtonsActions(
```

Comment:
function should have single responsibility in order to make your code clearer. Think of a way of splitting this function. And you should avoid having a lot of args inside the function